### DISCUSSION TITLE
Troubles with Emails & Messages app

### DISCUSSION THREAD

**Discussion by ab882** [saleor/apps#432] [Q&A]
Has anyone ran into issues setting up the Emails & Messages app? I've configured the app to use Gmail with SMTP settings from [https://support.google.com/mail/answer/7126229?hl=en#zippy=%2Cstep-change-smtp-other-settings-in-your-email-client](https://support.google.com/mail/answer/7126229?hl=en#zippy=%2Cstep-change-smtp-other-settings-in-your-email-client) but I can't seem to get any emails to send. 

Any pointers would be appreciated.

**Comment by krzysztofwolski** [comment]
GMail might be blocking sending emails due to it's anti spam measures. At least that was the issue when tried to use my gmail with the Sendgrid.

Are you looking for using gmail in production or just testing/staging? If the second option then I can recommend using mailtrap.io . Used it with the app previously with success.

I'll be able to look at gmail issue later today. If you'll be able to get any error messages/logs, it would help :raised_hands:

**Comment by ab882** [comment]
Hi, I'm just doing some testing at the moment. The plan is to go with a Sendgrid integration once we're ready for production. I've also set up a Sendgrid account and tried to use the Sendgrid SMTP settings to no avail so it could be an issue I'm having with SMTP in general? For what it's worth I'm using Saleor Cloud too...

**Reply by lkostrowski** [reply]
Hi @ab882 , if you plan to use Sendgrid anyway, its probably better to ignore SMTP/MJML functionality and just connect Sendgrid in the app. You can create your testing environment on Sendgrid level and when you are ready for production, you can update your tokens/templates

**Reply by ab882** [reply]
Hi @lkostrowski I'm trying to set up Sendgrid now but that doesn't seem to be working. I suspect I might have a permission missing on the API keys. Can you advise what should be applied here? Is there anything else in the setup process I need to know?

**Reply by lkostrowski** [reply]
Please provide a reproduction

**Reply by ab882** [reply]
Sorry I never wrote back - I ended up uninstalling and re-installing the latest version of the app and is rectified itself. As an aside, is there a way to update apps to the latest version?
